integrations/appsflyer/src/main/kotlin/com/rudderstack/integration/kotlin/appsflyer/Utils.kt

```diff
@@ -0,0 +1,364 @@
+@file:Suppress("TooManyFunctions")
+
+package com.rudderstack.integration.kotlin.appsflyer
+
+import com.appsflyer.AFInAppEventParameterName
+import com.appsflyer.AFInAppEventType
+import com.rudderstack.sdk.kotlin.android.utils.getArray
+import com.rudderstack.sdk.kotlin.core.ecommerce.ECommerceEvents
+import com.rudderstack.sdk.kotlin.core.ecommerce.ECommerceParamNames
+import com.rudderstack.sdk.kotlin.core.internals.utils.empty
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.boolean
+import kotlinx.serialization.json.booleanOrNull
+import kotlinx.serialization.json.double
+import kotlinx.serialization.json.doubleOrNull
+import kotlinx.serialization.json.int
+import kotlinx.serialization.json.intOrNull
+import kotlinx.serialization.json.long
+import kotlinx.serialization.json.longOrNull
+
+internal const val CREATIVE = "creative"
+
+// Reserved keywords for filtering custom properties
+internal val TRACK_RESERVED_KEYWORDS = listOf(
+    ECommerceParamNames.QUERY, ECommerceParamNames.PRICE, ECommerceParamNames.PRODUCT_ID,
+    ECommerceParamNames.CATEGORY, ECommerceParamNames.CURRENCY, ECommerceParamNames.PRODUCTS,
+    ECommerceParamNames.QUANTITY, ECommerceParamNames.TOTAL, ECommerceParamNames.REVENUE,
+    ECommerceParamNames.ORDER_ID, ECommerceParamNames.SHARE_MESSAGE, CREATIVE,
+    ECommerceParamNames.RATING
+)
+
+private const val FIRST_PURCHASE = "first_purchase"
+private const val REMOVE_FROM_CART = "remove_from_cart"
+private const val AF_ORDER_ID = "af_order_id"
+private const val PRODUCT_ID = "product_id"
+private const val CATEGORY = "category"
+private const val QUANTITY = "quantity"
+
+/**
+ * Maps RudderStack event to AppsFlyer event name and properties
+ */
+internal fun mapEventToAppsFlyer(eventName: String, properties: JsonObject?): Pair<String, MutableMap<String, Any>> {
+    val appsFlyerEventProps = mutableMapOf<String, Any>()
+    val appsFlyerEventName = getAppsFlyerEventName(eventName, properties, appsFlyerEventProps)
+    return Pair(appsFlyerEventName, appsFlyerEventProps)
+}
+
+@Suppress("CyclomaticComplexMethod")
+private fun getAppsFlyerEventName(
+    eventName: String,
+    properties: JsonObject?,
+    appsFlyerEventProps: MutableMap<String, Any>
+): String {
+    return when (eventName) {
+        ECommerceEvents.PRODUCTS_SEARCHED -> handleProductsSearched(properties, appsFlyerEventProps)
```

Comment:
Method name suggests that it'll return the AppsFlyer event name. But I can see it's also processing something i.e., constructing appsflyer event properties. Could we break this into multiple steps/methods?